nemo_skills/pipeline/utils/generation.py

```diff
@@ -175,6 +175,112 @@ def get_remaining_jobs(cluster_config, output_dir, random_seeds, chunk_ids, reru
     return missing_jobs
 
 
+def separate_hydra_args(extra_arguments: str) -> tuple[str, str]:
+    """
+    Separate Hydra config args (--config-*, --cfg, --info, etc.) and
+    other Hydra flags from override args (++*).
+
+    Args:
+        extra_arguments: String containing mixed Hydra and override arguments
+
+    Returns:
+        Tuple of (hydra_config_args, override_args) as strings
+
+    Examples:
+        Empty input:
+            separate_hydra_args("") -> ("", "")
+
+        With config path/name and overrides:
+            separate_hydra_args(
+                "--config-path /cfg --config-name my_cfg ++inference.temperature=0.7"
+            ) -> (
+                " --config-path /cfg --config-name my_cfg",
+                " ++inference.temperature=0.7",
+            )
+
+        With Hydra flags that take values (space and equals forms):
+            separate_hydra_args("--cfg all --info=plugins ++x=1") -> (
+                " --cfg all --info=plugins",
+                " ++x=1",
+            )
+
+        With Hydra flags without values:
+            separate_hydra_args("--run --multirun ++y=2") -> (
+                " --run --multirun",
+                " ++y=2",
+            )
+
+        Mixed value formats and directories:
+            separate_hydra_args(
+                "--config-path=/a --config-dir /b ++foo=bar"
+            ) -> (
+                " --config-path=/a --config-dir /b",
+                " ++foo=bar",
+            )
+    """
+    hydra_config_args = ""
+    override_args = ""
+
+    if not extra_arguments:
+        return hydra_config_args, override_args
+
+    # Hydra flags that do not take a value
+    hydra_flags_no_value = {
+        "--help",
+        "--hydra-help",
+        "--version",
+        "--resolve",
+        "--run",
+        "--multirun",
+        "--shell-completion",
+    }
+
+    # Hydra flags that take a value (accept both --flag=value and "--flag value")
+    hydra_flags_with_value = {
+        "--config-path",
+        "--config-name",
+        "--config-dir",
+        "--package",
+        "--experimental-rerun",
+        "--cfg",
+        "--info",
+    }
+
+    args_parts = extra_arguments.split()
```

Comment:
is this reliable? I imagine we could have cases when one of the parameter values might have a space inside? Not fully sure if this is going to cause problems

Reply:
good point. maybe `shlex.split` is more appropriate?